File: Hash Table/454. 4Sum II/4sum.py
class Solution:
    def fourSumCount(
        self, nums1: List[int], nums2: List[int], nums3: List[int], nums4: List[int]
    ) -> int:
        firstTwoSum = defaultdict(int)

        for i in nums1:
            for j in nums2:
                firstTwoSum[i + j] += 1

        count = 0

        for i in nums3:
            for j in nums4:
                count += firstTwoSum[-(i + j)]

        return count

File: Hash Table/454. 4Sum II/4sum.cpp
class Solution
{
public:
    int fourSumCount(vector<int> &nums1, vector<int> &nums2, vector<int> &nums3, vector<int> &nums4)
    {
        unordered_map<int, int> firstTwoSum;

        for (auto i : nums1)
        {
            for (auto j : nums2)
            {
                firstTwoSum[i + j]++;
            }
        }

        int count = 0;

        for (auto i : nums3)
        {
            for (auto j : nums4)
            {
                count += firstTwoSum[-(i + j)];
            }
        }
        return count;
    }
};